tests/users.spec.ts

```diff
@@ -0,0 +1,118 @@
+import {
+    describe, test, expect
+} from '@jest/globals';
+import {UsersService} from "./services/userService.ts";
+import {generateUserData} from "./helpers/user-data-generator.ts";
+import {IUser} from "./types.ts";
+import {UserRepository} from "./repositories/userRepository.ts";
+import {DBClient} from "./db-client/DBClient.ts";
+import {Client} from "ts-postgres";
+import {uuid} from "../src/utils.ts";
+
+const usersService = new UsersService();
+let client: Client;
+let userRepository: UserRepository;
+
+beforeAll(async () => {
+    client = await DBClient.createAndConnectClient();
+    userRepository = new UserRepository(client);
+});
+
+afterAll(async () => {
+    await client.end();
+});
+
+
+describe('Users endpoints', () => {
+    test('/users POST Should create a new user', async () => {
+        const userData = generateUserData();
+        const newUserResponse = await usersService.createUser(userData);
+        const createdUserData = newUserResponse.data.data;
+
+        expect(newUserResponse.status).toBe(200);
+        expect(createdUserData.id).toBeDefined();
+        expect(createdUserData.username).toEqual(userData.username);
+
+        const createdUserFromDB = await userRepository.selectOne(createdUserData.id);
+
+        if (createdUserFromDB !== null) {
```

Comment:
Lets have this as expect(createdUserFromDB).not.toBeNull(); or something like that.